File: app/src/play/java/me/ykrank/s1next/util/BuglyUtils.java
package me.ykrank.s1next.util;

import android.content.Context;
import android.support.annotation.NonNull;

import com.github.ykrank.androidtools.util.RxJavaUtil;
import com.tencent.bugly.crashreport.CrashReport;

import me.ykrank.s1next.BuildConfig;

import static me.ykrank.s1next.util.ErrorUtil.BUGLY_APP_ID;

/**
 * Created by ykrank on 2017/8/9.
 */

public class BuglyUtils {

    public static void init(@NonNull Context context) {
        final Context appContext = context.getApplicationContext();
        RxJavaUtil.workInRxIoThread(() -> {
            CrashReport.UserStrategy userStrategy = new CrashReport.UserStrategy(appContext);
            userStrategy.setAppVersion(BuildConfig.VERSION_NAME + "-" + BuildConfig.VERSION_CODE);
            CrashReport.initCrashReport(appContext, BUGLY_APP_ID, BuildConfig.DEBUG, userStrategy);
            CrashReport.setIsDevelopmentDevice(appContext, BuildConfig.DEBUG);
        });
    }
}

File: app/src/play/java/me/ykrank/s1next/util/BuglyUtils.kt
package me.ykrank.s1next.util

import android.content.Context
import com.github.ykrank.androidtools.util.RxJavaUtil
import com.tencent.bugly.crashreport.CrashReport
import com.tencent.bugly.crashreport.CrashReport.UserStrategy
import me.ykrank.s1next.BuildConfig
import me.ykrank.s1next.util.AppDeviceUtil.getAppFullVersionName

/**
 * Created by ykrank on 2017/8/9.
 */
object BuglyUtils {
    val isPlay: Boolean
        get() = true

    fun init(context: Context) {
        val appContext = context.applicationContext
        RxJavaUtil.workInRxIoThread {
            val userStrategy = UserStrategy(appContext)
            userStrategy.setAppVersion(getAppFullVersionName())
            CrashReport.initCrashReport(
                appContext,
                ErrorUtil.BUGLY_APP_ID,
                BuildConfig.DEBUG,
                userStrategy
            )
            CrashReport.setIsDevelopmentDevice(appContext, BuildConfig.DEBUG)
        }
    }
}
